ci-operator/templates/openshift/installer/cluster-launch-installer-upi-e2e.yaml

```diff
@@ -63,7 +63,7 @@ objects:
       name: ${JOB_NAME_SAFE}-bootstrap-exporter
     tls:
       termination: Edge
-      insecureEdgeTerminationPolicy: Redirect
+      insecureEdgeTerminationPolicy: Allow
```

Comment:
Redirecting means the connection between the cluster-under-test and the CI cluster will be secure when it carries real data.  I think we want that.  However, the fact that we're redirecting at all suggests the cluster-under-test is attempting an HTTP connection.  Can we fix that cluster to attempt an HTTPS connection?  That would protect the cluster-under-test from man-in-the-middle Ignition-config attacks.  The Edge termination gives us unencrypted communication inside the CI cluster even when we have HTTPS between the cluster-under-test and the CI ingress/router.

Reply:
I will grant you that doing it with HTTP exposes us to a man-in-the-middle attack. I don't care enough about that possibility, or the ramifications of said attack, when it comes to serving the bootstrap ignition config to the test cluster to prioritize that work at this juncture. If it is a big concern to you, then I am willing to take on that work post-4.1.

Reply:
I am willing to update the bootstrap ignition config to support a verification hash to combat a man-in-the-middle attack. That will require making changes to the upi terraform in the installer repo as well as changes here. Would you be comfortable accepting the changes in this PR to unblock e2e vsphere tests and pushing the verification hashes to follow-on PRs?

Reply:
/lgtm
/hold

I'm fine with this as a short-term fix, but I expected a `http` -> `https` change in the bootstrap node config to be all we need.  Is that not possible?  If not, feel free to pull the hold.

Reply:
Changing to https involves
1) generating certs,
2) modifying the terraform to add the certificate authority to the ignition config, and
3) changing the python http server to use ssl.

Reply:
> 1. generating certs,
> 2. modifying the terraform to add the certificate authority to the ignition config,

I thought Edge termination would have the CI cluster's ingress/router provision a new cert signed by a well-known CA.  Haven't tested it, but for example, https://origin-release.svc.ci.openshift.org/ is signed by DigiCert.

> 3. changing the python http server to use ssl, and
> 4. updating the route, service, and pod to use different ports.

Edge termination means the backing service can stick with HTTP.  The connection is:

    Boostrap Igntion -(HTTPS)-> CI-router, Edge termination -(HTTP)-> backing Service

Reply:
Ah, yes. Thanks for being patient with me on this and walking me through what should have been obvious.

/close